### PR TITLE
added load_data(), config, updated tests

### PR DESCRIPTION
**Pull request by taddyb**
# Added:
- `scripts` folder to store the CAMELS download script for simple runoff / precip vars
- updated `.gitignore`
- Added a `pyproj.toml` file in case we decide to be fancy
- Added a config object to keep track of variables using Pydantic
  - updated the model, and tests to reflect the config
- added a test for `load_data`

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+from pydantic import BaseModel, Field
+
+
+log = logging.getLogger(__name__)
+
+
+class Config(BaseModel):
+    input_size: int = 1
+    input_variable: List[str] = Field(default_factory=lambda: ['runoff_mm'])
```
Comment (hellkite500) [inline config.py:13]: Are these input and output variables backwards?

File: src/utils.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    data_dir = Path(__file__).parent.parent / 'data/CAMELS'
+    
+    # TODO add the input and output vars to config file
+    runoff_pd = pd.read_csv(data_dir / 'runoff_mm.csv')
+    precip_pd = pd.read_csv(data_dir / 'precipitation_mm.csv')
+    # attributes_pd = pd.read_csv(data_dir / 'attributes.csv')
+
+    # Calculating the mean for each basin across years.
+    runoff_mean = runoff_pd.mean().values[1:]
+    precip_mean = precip_pd.mean().values[1:]
```
Comment (hellkite500) [inline src/utils.py:27]: Likely worth a quick comment as to why the first value isn't used.

File: test/test_model.py [hunk 3]
```diff
@@ -1,49 +1,85 @@
...
+from src.utils import load_data
+
 
 @fixture
 def input():
     np.random.seed(42)
     data = np.random.uniform(0, 10, 24).reshape(24, 1)
     return torch.Tensor(data)
 
+@fixture
```
Comment (hellkite500) [inline test/test_model.py:19]: Probably should create a contest module for fixtures that span modules so the module test code can exist in independent test modules... Can be on the TODO list.

File: test/test_model.py [hunk 4]
```diff
@@ -1,49 +1,85 @@
...
-    model = Model(1,1, layers)
+    config.hidden_size = [10]
+    model = Model(config)
     #Pass input through the model
     tmp = model(input)
 
-@pytest.mark.usefixtures("input")
-def test_three_layer(input, compare):
+@pytest.mark.usefixtures("input", "config")
+def test_three_layer(config: Config, input, compare):
```
Comment (hellkite500) [inline test/test_model.py:66]: Not sure why this is duplicated. Based on the print, I can guess maybe it was a dev/debug function?

File: test/test_model.py [hunk 5]
```diff
@@ -52,7 +88,20 @@ def test_three_layer(input, compare):
...
+    """_summary_
+
+    Args:
+        data_dims (_type_): _description_
+    """
+    #Load the data
+    runoff_mean, precip_mean = load_data()
+    #Check the shapes
+    assert runoff_mean.shape == data_dims, "Runoff mean shape is incorrect"
+    assert precip_mean.shape == data_dims, "Precip mean shape is incorrect"
```
Comment (hellkite500) [inline test/test_model.py:107]: Add a new line for posix.

**Review by hellkite500 — COMMENTED** [review]
Doors project.toml need something in it to be useful?

**Comment by taddyb** [comment]
> Doors project.toml need something in it to be useful?

Not really. Some people use it in their repos for linting and pytest configurations. I can delete it until we need it

**Comment by hellkite500** [comment]
Will probably end up getting it populated at some point.  Will likely end up refactoring things into a package structure in the near future and we can add it then with the package metadata.

File: config.py (hunk 1 above)
Comment (taddyb) [reply config.py:13]: I think I committed an old Config.py instance by mistake. I removed the input and output vars from this file to reduce complexity

File: test/test_model.py (hunk 5 above)
Comment (taddyb) [reply test/test_model.py:62]: I added a check for the data type using `isinstance`. Is that what you meant by posix?

Comment (hellkite500) [reply test/test_model.py:62]: No, quite literally just a new line at the end of the file...[for quick explanation](https://www.baeldung.com/linux/files-end-with-newlines#:~:text=Newlines%20at%20the%20End%20of,defined%20as%20consisting%20of%20lines.)

File: src/utils.py (hunk 2 above)
Comment (hellkite500) [reply src/utils.py:28]: probably better to slice before the `mean()` aggregation then if the mean of the year isn't meaninful/needed.

File: test/test_model.py (hunk 5 above)
Comment (taddyb) [reply test/test_model.py:62]: done 👍 Thanks for the link/reading

**Review by hellkite500 — COMMENTED** [review]
Gonna leave this open in case you want to add newlines and consider the `mean` slicing change.  Those are minor and can be pushed on the todo stack, or I may push a commit or two to this branch to patch if I get back to more work on this in the near future.

File: src/utils.py (hunk 2 above)
Comment (taddyb) [reply src/utils.py:28]: This was how the code was implemented in the Colab notebook, but I agree it's not the best practice to slice after an operation. We can change this later and add as a TODO

**Pull request merged**